neuraxle/pipeline.py

```diff
@@ -486,3 +490,222 @@ def should_resume(self, data_container: DataContainer) -> bool:
+
+        return data_container
+
+    def _handle_fit_transform_sub_pipeline(self, sub_pipeline, data_container) -> \
+            Tuple['MiniBatchSequentialPipeline', DataContainer]:
+        """
+        Fit Transform sub pipeline using join fit transform.
+
+        :param sub_pipeline: sub pipeline to be used to transform data container
+        :param data_container: data container to fit transform
+        :return: fitted self, transformed data container
+        """
+        _, data_inputs = sub_pipeline.join_fit_transform(
+            data_inputs=data_container.data_inputs,
+            expected_outputs=data_container.expected_outputs
+        )
+        data_container.set_data_inputs(data_inputs)
+
+        current_ids = self.hash(data_container.current_ids, self.hyperparams, data_inputs)
+        data_container.set_current_ids(current_ids)
+
+        return self, data_container
+
+    def _create_sub_pipelines(self) -> List['MiniBatchSequentialPipeline']:
+        """
+        Create sub pipelines by splitting the steps by the join type name.
+
+        :return: list of sub pipelines
+        """
+        sub_pipelines: List[MiniBatchSequentialPipeline] = self.split(BARRIER_STEP_NAME)
+        for sub_pipeline in sub_pipelines:
+            if not sub_pipeline.ends_with_type_name(BARRIER_STEP_NAME):
+                raise Exception(
+                    'At least one Barrier step needs to be at the end of a streaming pipeline. '.format(
+                        self.join_type_name)
+                )
+
+        return sub_pipelines
+
+    def join_transform(self, data_inputs: Iterable) -> Iterable:
+        """
+        Concatenate the transform output of each batch of self.batch_size together.
+
+        :param data_inputs:
+        :return:
+        """
+        outputs = []
+        for batch in convolved_1d(
+                stride=self.batch_size,
+                iterable=data_inputs,
+                kernel_size=self.batch_size
+        ):
+            batch_outputs = super().transform(batch)
+            outputs.extend(batch_outputs)  # TODO: use a joiner here
+
+        return outputs
+
+    def join_fit_transform(self, data_inputs: Iterable, expected_outputs: Iterable = None) -> \
+            Tuple['MiniBatchSequentialPipeline', Iterable]:
+        """
+        Concatenate the fit transform output of each batch of self.batch_size together.
+
+        :param data_inputs: data inputs to fit transform on
+        :param expected_outputs: expected outputs to fit
+        :return: fitted self, transformed data inputs
+        """
+        outputs = []
+        for batch in convolved_1d(
+                stride=self.batch_size,
+                iterable=zip(data_inputs, expected_outputs),
+                kernel_size=self.batch_size
+        ):
+            di_eo_list = list(zip(*batch))
+            _, batch_outputs = super().fit_transform(list(di_eo_list[0]), list(di_eo_list[1]))
```

Comment:
Please name things when it's free like here: 
```
        for batch in convolved_1d(
                stride=self.batch_size,
                iterable=zip(data_inputs, expected_outputs),
                kernel_size=self.batch_size
        ):
                batch_data_inputs, batch_expected_outputs = list(zip(*batch))
```

Could we even do this, or something else cleaner?: 
```
        for batch_data_inputs, batch_expected_outputs in zip(convolved_1d(
                stride=self.batch_size,
                iterable=zip(data_inputs, expected_outputs),
                kernel_size=self.batch_size
        )):
```

Reply:
I did something better :)